### PR TITLE
Adding minified view and ability to hide "Show Options"

### PR DESCRIPTION
**Pull request by pashashocky**
SCR-20231013-nsfo

Allows for the above view :)

```
~~~dataviewjs
await dv.view("obsidian-feeds", {
	searchFor: "#log/flying",
	hideHeaders: true,
	showOptions: false,
})
~~~
```

### CODE REVIEW

File: view.js [hunk 1]
```diff
@@ -265,10 +272,22 @@ const result = dv
   .flatMap(state.showTree ? tree : (l) => (showParent(l) ? [l] : l.children))
   .filter((l) => !state.onlyWithTasks || someOfMeAndMyChildren(l, showTask))
   .groupBy((l) => (state.groupBySection ? l.link : l.link.toFile()))
-  .sort((g) => (state.sortByPath ? g : g.key.fileName()), "desc");
+  .sort((g) => (state.sortByPath ? g.key.fileName() : g), "desc")
```
Comment (lukemt) [inline view.js]: ?

Comment (lukemt) [reply view.js]: Maybe you thought this was a mistake, but it wasn't. If you don't call fileName, then it will sort by file path, instead of just the file name

File: view.js [hunk 2]
```diff
@@ -175,10 +182,10 @@ const tree = (listItem) => {
 
 const showParent = (listItem) => {
   if (listItem.section.subpath === fileName) return true;
-
-  const hasText = /[a-zA-Z0-9]/g.test(
-    listItem.text.replace(/\[\[[^\]]+\]\]/g, "")
-  );
+  const cleanText = listItem.text
```
Comment (lukemt) [inline view.js:185]: Why did you add cleanText? Where is it used?

File: view.js [hunk 3]
```diff
@@ -265,10 +272,22 @@ const result = dv
...
+    .flatMap((section) =>
+      section.rows.map((row) => ({
+        ...row,
+        text: `${section.key} - ${row.text}`,
+      }))
+    );
+  dv.taskList(result, false);
+} else {
+  dv.taskList(result);
+}
```
Comment (lukemt) [inline view.js:293]: Not tested, but this looks quite good

File: view.js [hunk 4]
```diff
@@ -4,11 +4,13 @@ const config = {
   excludeFolders: ["Logseq/logseq"],
   includeFolders: [],
   oneliners: true,
+  showOptions: true,
```
Comment (lukemt) [inline view.js:7]: Yeah I thought of it too! Naming things is always a strange endeavor though lol

**Review by lukemt — CHANGES_REQUESTED** [review]
Thank you for PR!

File: view.js (hunk 4 above)
Comment (lukemt) [reply view.js:7]: But is good i think

File: view.js (hunk 2 above)
Comment (pashashocky) [reply view.js:185]: I was using it for debugging `console.log({text: listItem.text, cleanText})`...
Easier to see what happens when messing around with regex replaces - it's only used in `const hasText` next statement.

File: view.js (hunk 1 above)
Comment (pashashocky) [reply view.js]: So, this is interesting - I'm happy to revert this change but the way it was originally it did not sort my daily feed in desc chronological order, `2027` would be the second entry, `2023` first, etc... 

**edit: I think it's because I hadn't wiped my dataview cache and moved files around!**

Why do you think it should be as before?

File: view.js (hunk 3 above)
Comment (pashashocky) [reply view.js:293]: I wasn't sure how to make that conditionally chained lol...

File: view.js (hunk 1 above)
Comment (lukemt) [reply view.js]: Because it would be wrong this way

Comment (lukemt) [reply view.js]: Path means folder + filename. Filename means only filename

File: view.js (hunk 2 above)
Comment (lukemt) [reply view.js:185]: My apologies, I was looking at it on the phone and didn't read it quite well. this looks perfect!

File: view.js (hunk 3 above)
Comment (lukemt) [reply view.js:293]: i donn't think this is a bad solution!

**Review by lukemt — APPROVED** [review]
Yey!
